examples/ASP.NET Core/README.md

```diff
@@ -0,0 +1,118 @@
+# OpenAI ASP.NET Core Example
+
+This example demonstrates how to use the OpenAI .NET client library with ASP.NET Core's dependency injection container, registering a ChatClient as a singleton for optimal performance and resource usage.
+
+## Features
+
+- **Singleton Registration**: ChatClient registered as singleton in DI container
+- **Thread-Safe**: Demonstrates concurrent usage for chat completion endpoints
+- **Configurable Model**: Model selection via configuration (appsettings.json)
+- **Modern ASP.NET Core**: Uses minimal APIs with async/await patterns
+
+## Prerequisites
+
+- .NET 8.0 or later
+- OpenAI API key
+
+## Setup
+
+1. **Set your OpenAI API key** using one of these methods:
+
+   **Environment Variable (Recommended):**
```

Comment:
silly as it may seem, we should consider using Windows conventions here by default (or PowerShell) and linking out to docs for *nix shells.   For some reason, we've seen a number of folks struggle with using environment variables in the .NET ecosystem.